sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubProducerClient.cs

```diff
@@ -391,6 +394,44 @@ public virtual async Task<string[]> GetPartitionIdsAsync(CancellationToken cance
             return await Connection.GetPartitionPropertiesAsync(partitionId, RetryPolicy, cancellationToken).ConfigureAwait(false);
         }
 
+        /// <summary>
+        ///   A set of information about the state of publishing for a partition, as observed by the <see cref="EventHubProducerClient" />.  This
+        ///   data can always be read, but will only be populated with information relevant to the features which are active for the producer client.
+        /// </summary>
+        ///
+        /// <param name="partitionId">The unique identifier of a partition associated with the Event Hub.</param>
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken" /> instance to signal the request to cancel the operation.</param>
+        ///
+        /// <returns>The set of information about the publishing state of the requested partition, within the context of this producer.</returns>
+        ///
+        public virtual async Task<PartitionPublishingProperties> ReadPartitionPublishingPropertiesAsync(string partitionId,
+                                                                                                        CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotClosed(IsClosed, nameof(EventHubProducerClient));
+            Argument.AssertNotNullOrEmpty(partitionId, nameof(partitionId));
+
+            var partitionState = PartitionState.GetOrAdd(partitionId, new PartitionPublishingState(partitionId));
+            cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+
+            try
+            {
+                await partitionState.PublishingGuard.WaitAsync(cancellationToken).ConfigureAwait(false);
+
+                if (!partitionState.IsInitialized)
+                {
+
+                    cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+                    await InitializePartitionStateAsync(partitionState, cancellationToken).ConfigureAwait(false);
+                }
+
+                return CreatePublishingPropertiesFromPartitionState(Options, partitionState);
```

Comment:
I'm wondering whether class `PartitionPublishingProperties` is needed because we can expose the properties from PartitionPublishingState directly. I'm going to ask the same question to my fellow Java developers.

Reply:
The state is an internal construct that is mutable and has a synchronization primitive.   We want to return a snapshot of the state, not allow external callers access to the mutable version.

Reply:
This is a good point. Giving users a light-weight snapshot is better because they might pass it around.